src/components/verification-grid/verification-grid.ts

```diff
@@ -0,0 +1,1364 @@
+    // the decision history as a new decision
+    this.decisions.push(...decisions);
+    this.dispatchEvent(new CustomEvent(VerificationGridComponent.decisionMadeEventName, { detail: decisions }));
+
+    if (this.autoPage) {
+      this.showDecisionButtonHighlight([event.target as any]);
+      this.createSelectionHighlight(selectedTiles, decisionColor);
+
+      // we wait for 300ms so that the user has time to see the decision that
+      // they have made in the form of a decision highlight around the selected
+      // grid tiles and the chosen decision button
+      await sleep(300);
+      this.removeDecisionHighlight(selectedTiles);
+
+      this.nextPage(decisions.length);
+    }
+  }
+
+  private setDecisionDisabled(disabled: boolean): void {
+    const decisionElements = this.decisionElements ?? [];
+    for (const decisionElement of decisionElements) {
+      decisionElement.disabled = disabled;
+    }
+  }
+
+  //#endregion
+
+  //#region DecisionHighlights
+
+  private decisionColor(verification: DecisionWrapper): string {
+    // TODO: this is really slow, we should probably optimize it
+    const decisionElements = this.decisionElements ?? [];
+    const decisionButton = decisionElements.find((element: DecisionComponent) => {
+      const verificationDecisions = verification.decisions;
+      const elementModels = element.classificationModels();
+      return JSON.stringify(verificationDecisions) === JSON.stringify(elementModels);
+    });
+
+    if (!decisionButton) {
+      throw new Error("Could not find color to match decision");
+    }
+
+    return decisionButton.color;
+  }
+
+  private createSelectionHighlight(selectedTiles: VerificationGridTileComponent[], color: string | string[]): void {
+    selectedTiles.forEach((tile: VerificationGridTileComponent, i: number) => {
+      const derivedColor = Array.isArray(color) ? color[i] : color;
+      tile.borderColor = derivedColor;
+    });
+  }
+
+  private createDecisionHighlight(decisions: DecisionWrapper[]): void {
+    decisions.forEach((decision: DecisionWrapper, i: number) => {
+      const color = this.decisionColor(decision);
+      this.gridTiles[i].borderColor = color;
+    });
+  }
+
+  private removeDecisionHighlight(selectedTiles: VerificationGridTileComponent[] = Array.from(this.gridTiles)): void {
+    for (const tile of selectedTiles) {
+      tile.borderColor = "var(--oe-panel-color)";
+    }
+
+    this.removeDecisionButtonHighlight();
+  }
+
+  private touchedDecisionElements(): DecisionComponent[] {
+    // TODO: this is really inefficient. We should improve the perf and quality
+    const decisionElements = this.decisionElements ?? [];
+    return decisionElements.filter((element: DecisionComponent) => {
+      const possibleDecisions = element.classificationModels();
+      return this.decisions.some((decision: DecisionWrapper) => {
+        return JSON.stringify(decision.decisions) === JSON.stringify(possibleDecisions);
```

Comment:
do not do structural quality this way

i'm not sure what this method is doing?

Reply:
This method was returning an array of decision buttons that were used to make decisions.

I used this later on so that I could show the decision button highlights only on the decisions that were used